### PR TITLE
Update mule docs for new pipeline

### PR DESCRIPTION
**Pull request by btoll**
## Summary

Update `mule` docs for the new pipeline.

The changes in the other scripts are to reflect what appears in the docs, i.e.:

- `ARCH` becomes `ARCH_TYPE`
- `VER` or `FULLVERSION` becomes `VERSION`

Also, futzed with jobs in `package.yaml`.

## Test Plan

n/a

### CODE REVIEW

File: scripts/release/mule/README.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+algorand-devtools_dev_linux-amd64_2.1.86615.deb
+```
+
+Note that this is in the format `{ALGORAND_PACKAGE_NAME}_{CHANNEL}_{OS_TYPE}-{ARCH_TYPE}_{VERSION}.deb. `rpm` packages will follow their own format which is easy to intuit.
+
+It is common that a custom build is performed on a feature branch other than `rel/stable` or `rel/beta` and that the build environment will need to be modified. In these instances, it is important to be able to pass values to the build process to customize a build.
+
+The most common way to do this is to modify the environment that the subprocess inherits by specifying the values on the command *before* the command.  This won't need to be done for the package stage, but often needs to be done with subsequent stages.
+
+In order to be able to correctly run some of the stages, such as testing and signing, several values needed by the subsequent stages must to be explicitly passed to those stages.
```
Comment (egieseke) [inline scripts/release/mule/README.md:134]: "must be explicitly passed..."

File: scripts/release/mule/package/rpm/package.sh [hunk 2]
```diff
@@ -5,12 +5,12 @@ set -ex
...
-ARCH=$(./scripts/archtype.sh)
+ARCH_TYPE=$(./scripts/archtype.sh)
 OS_TYPE=$(./scripts/ostype.sh)
-FULLVERSION=${VERSION:-$(./scripts/compute_build_number.sh -f)}
+VERSION=${VERSION:-$(./scripts/compute_build_number.sh -f)}
 BRANCH=${BRANCH:-$(git rev-parse --abbrev-ref HEAD)}
 CHANNEL=${CHANNEL:-$(./scripts/compute_branch_channel.sh "$BRANCH")}
-ALGO_BIN="$REPO_DIR/tmp/node_pkgs/$OS_TYPE/$ARCH/$CHANNEL/$OS_TYPE-$ARCH/bin"
+ALGO_BIN="$REPO_DIR/tmp/node_pkgs/$OS_TYPE/$ARCH_TYPE/$CHANNEL/$OS_TYPE-$ARCH_TYPE/bin"
 # TODO: Should there be a default network?
```
Comment (egieseke) [inline scripts/release/mule/package/rpm/package.sh:14]: Is there a resolution to this question?

**Review by egieseke — DISMISSED** [review]

File: scripts/release/mule/README.md (hunk 1 above)
Comment (btoll) [reply scripts/release/mule/README.md:134]: Will fix.

File: scripts/release/mule/package/rpm/package.sh (hunk 2 above)
Comment (btoll) [reply scripts/release/mule/package/rpm/package.sh:14]: Ha.  My feeling is probably?

Comment (btoll) [reply scripts/release/mule/package/rpm/package.sh]: I removed the comment.

File: scripts/release/mule/README.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+algorand-devtools_dev_linux-amd64_2.1.86615.deb
+```
+
+Note that this is in the format `{ALGORAND_PACKAGE_NAME}_{CHANNEL}_{OS_TYPE}-{ARCH_TYPE}_{VERSION}.deb. `rpm` packages will follow their own format which is easy to intuit.
+
+It is common that a custom build is performed on a feature branch other than `rel/stable` or `rel/beta` and that the build environment will need to be modified. In these instances, it is important to be able to pass values to the build process to customize a build.
+
+The most common way to do this is to modify the environment that the subprocess inherits by specifying the values on the command *before* the command.  This won't need to be done for the package stage, but often needs to be done with subsequent stages.
+
+In order to be able to correctly run some of the stages, such as testing and signing, several values needed by the subsequent stages must be explicitly passed to those stages.
```
Comment (bricerisingalgorand) [inline scripts/release/mule/README.md:139]: several values are needed*

(I'm not sure how to do the suggest changes thing that John does)

**Review by bricerisingalgorand — APPROVED** [review]

**Review by onetechnical — APPROVED** [review]
I'm not wild about the bash expansions (`-x`) because our logging is a little wordy, but otherwise it seems okay.

**Pull request merged**